docs/rules/no-underscore-dangle.md

```diff
@@ -33,17 +33,21 @@ var _ = require('underscore');
 var obj = _.contains(items, item);
 obj.__proto__ = {};
 var file = __filename;
+function foo(_bar) {};
+const foo = { onClick(_bar) {} };
+const foo = (_bar) => {};
 ```
 
 ## Options
 
 This rule has an object option:
 
-* `"allow"` allows specified identifiers to have dangling underscores
-* `"allowAfterThis": false` (default) disallows dangling underscores in members of the `this` object
-* `"allowAfterSuper": false` (default) disallows dangling underscores in members of the `super` object
-* `"allowAfterThisConstructor": false` (default) disallows dangling underscores in members of the `this.constructor` object
-* `"enforceInMethodNames": false` (default) allows dangling underscores in method names
+-  `"allow"` allows specified identifiers to have dangling underscores
+-  `"allowAfterThis": false` (default) disallows dangling underscores in members of the `this` object
+-  `"allowAfterSuper": false` (default) disallows dangling underscores in members of the `super` object
+-  `"allowAfterThisConstructor": false` (default) disallows dangling underscores in members of the `this.constructor` object
+-  `"enforceInMethodNames": false` (default) allows dangling underscores in method names
+-  `"allowFunctionParams": true` (default) allows dangling underscores in function parameter names
```

Comment:
To make more sense, it should defaults to `false`. we can change it in next major release( as it's a breaking change). 😄